x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/job_messages_pane.tsx

```diff
@@ -6,25 +6,40 @@
  */
 
 import React, { FC, useCallback, useEffect, useState } from 'react';
+import { EuiButton, EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiToolTip } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
 import { ml } from '../../../../services/ml_api_service';
 import { JobMessages } from '../../../../components/job_messages';
 import { JobMessage } from '../../../../../../common/types/audit_message';
 import { extractErrorMessage } from '../../../../../../common/util/errors';
 import { useToastNotificationService } from '../../../../services/toast_notification_service';
+import { useMlKibana } from '../../../../../application/contexts/kibana';
 interface JobMessagesPaneProps {
   jobId: string;
   start?: string;
   end?: string;
   actionHandler?: (message: JobMessage) => void;
+  refreshJobList?: () => void;
 }
 
 export const JobMessagesPane: FC<JobMessagesPaneProps> = React.memo(
-  ({ jobId, start, end, actionHandler }) => {
+  ({ jobId, start, end, actionHandler, refreshJobList }) => {
     const [messages, setMessages] = useState<JobMessage[]>([]);
     const [isLoading, setIsLoading] = useState(false);
     const [errorMessage, setErrorMessage] = useState('');
+    const [isClearing, setIsClearing] = useState<boolean>(false);
+
     const toastNotificationService = useToastNotificationService();
+    const {
+      services: {
+        mlServices: {
+          mlApiServices: {
+            jobs: { clearJobAuditMessages },
+          },
+        },
+      },
+    } = useMlKibana();
```

Comment:
`useMlApiContext` is a nicer context provider for the api functions

Reply:
Updated in 6e82c56ae7f29538bc0170a90e54d8b6f16d24a3